KOGITO-3797 Avoid timeout exception from Infinispan

cristianonicolai opened the pull request:
https://issues.redhat.com/browse/KOGITO-3797

tiagodolphine reviewed (approved):
LGTM, is it worth adding the configs to jobs-service?

cristianonicolai commented:
@tiagodolphine I cant reproduce the same issue on Job service, and just realized that adding these properties hasnt actually fixed the issue, still investigating.

cristianonicolai commented:
@tiagodolphine @MarianMacik please take another look as I changed the solution completly. Basically seems that the issue is due to try to create a cache with a template name that doesnt exist. I've changed that, so now the user has to define a template name if he wants to use one.
@ricardozanini @radtriste  FYI, not sure if this has any impact on operator or BDD tests.

radtriste commented on persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/cache/InfinispanCacheManager.java:
```diff
@@ -72,42 +74,39 @@ public void destroy() {
-            throw e;
+    protected <K, V> RemoteCache<K, V> getOrCreateCache(final String name) {
+        LOGGER.debug("Trying to get cache {} from the server", name);
+        RemoteCache<K, V> remoteCache = manager.getCache(name);
+        return remoteCache == null ? createCache(name) : remoteCache;
```
why not using the `Optional.of().orElse` ?

cristianonicolai replied:
old style coder? :P dont see much of a need for that in here to be honest.

radtriste replied:
```suggestion
        return Optional.ofNullable(manager.getCache(name)).orElse(createCache(name))
```
:P

radtriste commented on persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/cache/InfinispanCacheManager.java:
```diff
@@ -72,42 +74,39 @@ public void destroy() {
+
+    protected <K, V> RemoteCache<K, V> createCache(final String name) {
+        RemoteCacheManagerAdmin admin = manager.administration();
+        if (cacheTemplateName.isPresent()) {
+            LOGGER.debug("Cache {} not found, trying to create a new one based on template {}", name, cacheTemplateName.get());
+            return admin.createCache(name, cacheTemplateName.get());
+        } else {
+            LOGGER.debug("Cache {} not found, trying to create a new one based on Kogito default configuration", name);
+            return admin.createCache(name, new KogitoCacheDefaultConfiguration(name));
```
same here, something like

```
return admin.createCache(name, cacheTemplate.orElse(new KogitoCacheDefaultConfiguration(name)))
```

Also the DEBUG log should not be there IMO or change the message `Cache {} not found` => `Creating cache based on ...`

cristianonicolai replied:
ok, I tweaked the log because I think it can be handy to know which config will be used, at least helped me during the tests.

radtriste reviewed (approved):
minor comments

ricardozanini commented:
@cristianonicolai should not be a problem on the operator side.

cristianonicolai commented:
jenkins please retest this

MarianMacik reviewed (approved):
Looks good to me. Thanks!

cristianonicolai commented:
jenkins please retest this

cristianonicolai commented:
jenkins please retest this
